### PR TITLE
Add basic docstrings to some functions

### PR DESCRIPTION
**Pull request by JSorngard**
This PR adds basic docstrings for some of the functions in the crate.

Related to #336.

### CODE REVIEW

File: src/math/j0.rs [hunk 1]
```diff
@@ -109,6 +109,7 @@ const S02: f64 = 1.16926784663337450260e-04; /* 0x3F1EA6D2, 0xDD57DBF4 */
 const S03: f64 = 5.13546550207318111446e-07; /* 0x3EA13B54, 0xCE84D5A9 */
 const S04: f64 = 1.16614003333790000205e-09; /* 0x3E1408BC, 0xF4745D8F */
 
+/// Zeroth order of the Bessel function of the first kind (f64).
```
Comment (tgross35) [inline src/math/j0.rs:112]: I think bessel functions probably aren't common knowledge, could you add a link to https://en.wikipedia.org/wiki/Bessel_function?

**Review by tgross35 — COMMENTED** [review]
Thanks! One request but mostly lgtm.

File: src/math/j0.rs (hunk 1 above)
Comment (JSorngard) [reply src/math/j0.rs:112]: Sure!

Comment (JSorngard) [reply src/math/j0.rs]: Done! Is this good?

**Review by tgross35 — APPROVED** [review]
Thank you!

**Pull request merged**